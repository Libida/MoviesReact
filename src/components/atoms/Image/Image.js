import React from "react";
import propTypes from "prop-types";
import {NO_IMG_URL} from "../../../constants/strings";

export default function Image({src, alt, title, incomeClasses="", incomeWrapClasses=""}) {
    const isNoImg = (src === NO_IMG_URL);
    let baseClasses = "img";

    if (isNoImg) {
        baseClasses+= " img--no-img";
    }

    return (
        <span className={`img-wrap ${incomeWrapClasses}`}>
            <img src={src || NO_IMG_URL} alt={alt || title} title={title} className={`${baseClasses} ${incomeClasses}`}/>
        </span>
    );
};

Image.propTypes = {
    src: propTypes.string,
    classes: propTypes.string
};

Image.defaultProps = {
    src: NO_IMG_URL
};