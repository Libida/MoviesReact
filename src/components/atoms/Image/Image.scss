.img-wrap {
  display: block;
  position: relative;
  outline: none;
  width: 100%;
}
