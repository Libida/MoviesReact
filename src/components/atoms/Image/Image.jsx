import React from "react";
import propTypes from "prop-types";
import {NO_IMG_URL} from "../../../constants/strings";
import "./Image.scss";

export function Image({src, alt, title, incomeClasses="", incomeWrapClasses=""}) {
    const isNoImg = (src === NO_IMG_URL);

    return (
        <span className={`img-wrap ${incomeWrapClasses}`}>
            <img src={src || NO_IMG_URL} alt={alt || title} title={title} className={`img ${isNoImg ? "img--no-img" : ""} ${incomeClasses}`}/>
        </span>
    );
};

Image.propTypes = {
    src: propTypes.string,
    classes: propTypes.string
};

Image.defaultProps = {
    src: NO_IMG_URL
};