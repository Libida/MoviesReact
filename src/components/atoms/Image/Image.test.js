import React from "react";
import renderer from "react-test-renderer";
import Image from "./Image";

it("simple image renders correctly", () => {
    const tree = renderer
        .create(<Image src="https://image.tmdb.org/t/p/w500/wB58wlBAr6484Wm6VyFDqSD8VOJ.jpg" title="image"/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it("image without src renders correctly", () => {
    const tree = renderer
        .create(<Image title="image"/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});