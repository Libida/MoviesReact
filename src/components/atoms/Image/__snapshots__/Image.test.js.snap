// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`image without src renders correctly 1`] = `
<span
  className="img-wrap "
>
  <img
    alt="image"
    className="img img--no-img "
    src="https://sciences.ucf.edu/psychology/wp-content/uploads/sites/63/2019/09/No-Image-Available.png"
    title="image"
  />
</span>
`;

exports[`simple image renders correctly 1`] = `
<span
  className="img-wrap "
>
  <img
    alt="image"
    className="img  "
    src="https://image.tmdb.org/t/p/w500/wB58wlBAr6484Wm6VyFDqSD8VOJ.jpg"
    title="image"
  />
</span>
`;
